master/internal/experiment.go

```diff
@@ -356,8 +356,15 @@ func (e *internalExperiment) TrialReportProgress(msg experiment.TrialReportProgr
 	e.mu.Lock()
 	defer e.mu.Unlock()
 
-	e.searcher.SetTrialProgress(msg.RequestID, msg.Progress)
-	progress := e.searcher.Progress()
+	progress := float64(msg.Progress)
+	if !msg.IsRaw {
+		e.searcher.SetTrialProgress(msg.RequestID, msg.Progress)
+		progress = e.searcher.Progress()
+	}
+	if progress < 0 || progress > 1 {
+		e.syslog.Errorf("Invalid progress value: %f", progress)
+		return nil
```

Comment:
Sure, but I'm curious why the DB error next line is ignored?

Reply:
it should also be returned, feel free to update it if you want. the only reason i can think of to not return it is if you have a one-off DB connection failure or something and you don't want to exit training because progress reporting isn't important. 

this case is important tho IMO because if progress isn't [0,1] then that means user code is wrong and they should know to fix it.